.github/workflows/ardionoCI.yml

```diff
@@ -0,0 +1,33 @@
+name: CI Arduino
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on: [push, pull_request]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build-sketch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone Repo
+        uses: actions/checkout@v2
```

Comment:
Get the extra dependency you need:

```yaml
      - name: Get GPS library
        uses: actions/checkout@v2
        with:
          repository: mikalhart/TinyGPSPlus
          path: external/TinyGPSPlus
```

And then later on use it when compiling.